Replace weighted_shuffle() with weighted_order()

ajyoon opened the pull request:
Follow up to [Issue 1](https://github.com/ajyoon/blur/issues/1) and commit [849f874](https://github.com/ajyoon/blur/commit/849f874c461e951116cc7bac790321ba3d09019d). The specification of `weighted_shuffle()` was not clear enough, and its use case was unclear.

Docstring of the new function:

``` python
def weighted_order(weights):
    """
    Non-uniformally order a list according to weighted priorities.

    ``weights`` is a list of tuples of the form ``(item, priority)`` of
    types ``(Any, float or int)``. The output list is constructed by repeatedly
    calling ``weighted_choice()`` on the weights, adding items to the end of
    the list as the are picked.

    Higher value weights will have a higher chance of appearing near the
    beginning of the output list.

    A list of all uniform weights is equivalent to calling ``random.shuffle()``
    on the list of values.

    Args:
        weights (list[(Any, float or int)]):

    Returns:
        list: the newly ordered list
    """
```

ajyoon committed:
Replace weighted_shuffle() with weighted_order() and update tests

ajyoon committed:
Remove debugging artifact

NathanC commented on blur/rand.py:
```diff
@@ -384,78 +384,35 @@ def weighted_choice(weights):
 
 
-def weighted_shuffle(weights):
+def weighted_order(weights):
```
maybe "weighted_sort"?

ajyoon replied:
"sort" makes me think that we are sorting relative to the values which appear in the final list. We are sorting, but not exactly. Maybe...

NathanC replied:
Isn't it a sort? You're sorting probabilistically based on weight.

NathanC replied:
I mean, sorting _is_ ordering. So I guess either is fine.

NathanC commented on blur/rand.py:
```diff
@@ -384,78 +384,35 @@ def weighted_choice(weights):
+        output_list.append(picked_value)
+        # Remove the picked item from remaining_items
+        remove_item = next((item for item in remaining_items
+                            if item[0] == picked_value))
```
won't this remove the first duplicate value for "picked_value" that it comes across? so if I have ("a", 20), ("b", 40), ("a", 10000) and I choose the last one in line 412, I'd end up removing the first one.

NathanC replied:
value equality is not identity

ajyoon committed:
Fix order ambiguity in weighted_order()

Also adds an option to weighted_choice() to return a 2-tuple
of (index, value) in addition to the existing option of returning
just the chosen value.

Also adds a ProbabilityUndefinedError exception to the module,
and implements it in weighted_choice() and weighted_order()

Tests have been updated to cover new behavior.
